agg-ast/schema_derivation/src/schema_derivation.rs

```diff
@@ -1146,7 +1146,25 @@ impl DeriveSchema for TaggedOperator {
                     set! {then_schema, else_schema},
                 )))
             }
-            TaggedOperator::Filter(_) => todo!(),
+            TaggedOperator::Filter(f) => {
+                let input_schema = f.input.derive_schema(state)?;
+                let is_nullable = input_schema.satisfies(&NULLISH.clone()) != Satisfaction::Not;
+                let array_schema = array_schema_or_error!(input_schema, f.input);
```

Comment:
```suggestion
                let array_element_schema = array_schema_or_error!(input_schema, f.input);
```
If I understand the macro correctly, this is getting the schema from within a `Schema::Array`. So that represents the schema of the elements of the array, right? Could we consider this rename for the variable, and even possibly changing the name of the macro (and adding a comment to it) to reflect that?

<hr>

Side note: is the `array_schema_or_error` macro complete? Why does it only take the first `Array` schema found in an `AnyOf` as opposed to collecting all possible array schemas? For example, what about something like `AnyOf([Array(Int), Array(String)])`?

Reply:
`AnyOf([Array(Int), Array(String)])` _should_ be impossible, but that's something we may want to audit. Simplify should definitely collapse them. Union already does.

Reply:
I know `ArrayOf([AnyOf([Int, String])])` is actually less precise, but that's one of those things that is fine to keep sizes down.

Reply:
Ooh ok yes I should have checked `simplify`. I had a feeling it would combine those into `Array(AnyOf([Int, String]))` but I decided to just not check and assume since that's a different definition we didn't do it. Since we do do that, though, that's totally fine. No reason to change the macro implementation!

I still think a name change for the macro and the variable could be useful for reading purposes.